lib/server.js

```diff
@@ -208,6 +209,38 @@ function listenPromise (server, listenOptions) {
   })
 }
 
+// This function used to compile a validation function for http version
+function compileValidateHTTPVersion (options) {
```

Comment:
This leading comment is not very useful. It would be better to provide a full jsdoc block:

```suggestion
/**
 * Creates a function, based upon initial configuration, that will
 * be used to verify every incoming request conforms to allowed
 * HTTP versions for the Fastify instance, e.g. a Fastify HTTP/1.1
 * server will not serve HTTP/2 requests upon the result of the
 * verification function.
 *
 * @param {object} options
 * @param {function} [options.serverFactory] If present, the
 * validator function will skip all checks.
 * @param {boolean} [options.http2 = false] <need a description>
 *
 * @returns {function} HTTP version validator function.
 */
function compileValidateHTTPVersion (options) {
```